src/main/java/core/basesyntax/Car.java

```diff
@@ -1,13 +1,98 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
 import java.util.List;
 
 /**
  * Make this class immutable. See requirements in the README file
  */
-public class Car {
-    private int year;
-    private String color;
-    private List<Wheel> wheels;
-    private Engine engine;
+public final class Car implements Cloneable {
+    private final int year;
+    private final String color;
+    private final List<Wheel> wheels;
+    private final Engine engine;
+
+    public Car(int year, String color, List<Wheel> wheels, Engine engine) {
+        this.year = year;
+        this.color = color;
+        this.wheels = wheelsPackage(wheels);
+        this.engine = installEngine(engine);
+    }
+
+    public int getYear() {
+        return year;
+    }
+
+    public Car changeYear(int year) {
+        return new Car(year, color, wheels, engine.clone());
```

Comment:
```suggestion
        return new Car(year, color, wheels, engine);
```
in the constructor you will make a copy of the `engine`

Reply:
done